pkg/volume/csi/csi_attacher.go

```diff
@@ -63,29 +65,22 @@ func (c *csiAttacher) Attach(spec *volume.Spec, nodeName types.NodeName) (string
 		return "", errors.New("missing spec")
 	}
 
-	csiSource, err := getCSISourceFromSpec(spec)
+	driverName, volumeHandle, _, _, err := getCSIFieldsFromSpec(spec)
 	if err != nil {
 		klog.Error(log("attacher.Attach failed to get CSI persistent source: %v", err))
 		return "", err
 	}
+	if err := c.plugin.assertVolumeHandle(driverName, volumeHandle); err != nil {
```

Comment:
This function will return success for ephemeral volume types, which could pass an empty volumeHandle down to `newVolumeAttachment`, which I think we don't want to support.  I suggest changing this function to only assert for persistent volume types.